# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationChild.h
# Commit: 1f3365d25033
# Full Hash: 1f3365d25033f9dbc52b13e8fc4c64f3cdd35946
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r cc25e5c94813 -r 1f3365d25033 dom/notification/NotificationChild.h
--- a/dom/notification/NotificationChild.h	Mon Nov 11 20:34:11 2024 +0000
+++ b/dom/notification/NotificationChild.h	Mon Nov 11 20:34:11 2024 +0000
@@ -7,11 +7,13 @@
 #ifndef DOM_NOTIFICATION_NOTIFICATIONCHILD_H_
 #define DOM_NOTIFICATION_NOTIFICATIONCHILD_H_
 
+#include "mozilla/WeakPtr.h"
 #include "mozilla/dom/notification/PNotificationChild.h"
 
 namespace mozilla::dom::notification {
 
-class NotificationChild final : public PNotificationChild {
+class NotificationChild final : public PNotificationChild,
+                                public SupportsWeakPtr {
   NS_INLINE_DECL_REFCOUNTING(NotificationChild)
 
  private: